samples/openthread/cli/common.cmake

```diff
@@ -0,0 +1,97 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+# Header guard
+if(__OT_CLI_COMMON_CMAKE__)
+  return()
+endif()
+set(__OT_CLI_COMMON_CMAKE__ TRUE)
+
+include(${NRFXLIB_DIR}/common.cmake)
+
+# Copy new OpenThread libraries, headers, text file with the configuration
+# to the nrfxlib openthread folder.
+function(ot_libs_substitute)
+  execute_process(
+    COMMAND           git rev-parse HEAD
+    WORKING_DIRECTORY ${openthread_SOURCE_DIR}
+    RESULT_VARIABLE   git_return
+    OUTPUT_VARIABLE   openthread_git_hash
+    )
+
+  list(APPEND OPENTHREAD_SETTINGS "OpenThread_commit=${openthread_git_hash}\n\n")
+
+  # Store all OT related variables
+  get_cmake_property(_variableNames VARIABLES)
+  list (SORT _variableNames)
+  foreach (_variableName ${_variableNames})
+    if("${_variableName}" MATCHES "^CONFIG_OPENTHREAD_.*|^OT_.*")
+      list(APPEND OPENTHREAD_SETTINGS "${_variableName}=${${_variableName}}\n")
+    endif()
+  endforeach()
+
+  set(OPENTHREAD_CONFIG_FILE
+    "${PROJECT_BINARY_DIR}/openthread_lib_configuration.txt")
+  FILE(WRITE ${OPENTHREAD_CONFIG_FILE} ${OPENTHREAD_SETTINGS})
+
+  set(OPENTHREAD_SRC_DIR "${PROJECT_BINARY_DIR}/modules/openthread/build/src")
+  set(OPENTHREAD_LIBS
+    "${OPENTHREAD_SRC_DIR}/core/libopenthread-ftd.a"
+    "${OPENTHREAD_SRC_DIR}/core/libopenthread-mtd.a"
+    "${OPENTHREAD_SRC_DIR}/core/libopenthread-radio.a"
+    "${OPENTHREAD_SRC_DIR}/cli/libopenthread-cli-ftd.a"
+    "${OPENTHREAD_SRC_DIR}/cli/libopenthread-cli-mtd.a"
+    "${OPENTHREAD_SRC_DIR}/ncp/libopenthread-ncp-ftd.a"
+    "${OPENTHREAD_SRC_DIR}/ncp/libopenthread-ncp-mtd.a"
+    "${OPENTHREAD_SRC_DIR}/ncp/libopenthread-rcp.a"
+  )
+
+  nrfxlib_calculate_lib_path(lib_path)
+
+  foreach(lib IN LISTS OPENTHREAD_LIBS)
+    add_custom_command(
+      TARGET app
+      COMMENT "Copying OpenThread library: ${lib}."
+      PRE_LINK
+      COMMAND ${CMAKE_COMMAND} -E copy "${lib}"
+        "${NRFXLIB_DIR}/openthread/${lib_path}/v${OT_THREAD_VERSION}"
```

Comment:
This is not copying as expected:

Result:
```
$ file openthread/lib/cortex-m4/soft-float/v1.1 
openthread/lib/cortex-m4/soft-float/v1.1: current ar archive
```

Reply:
I will check that, @edmont reported some copying issues in his tests as well.

Reply:
Copying works now both for v1.1 and v.1.2. I but we need to hold this PR till architecture is changed to meet behavior described above.